dukan/src/main/kotlin/net/thechance/dukan/repository/DukanRepository.kt

```diff
@@ -4,14 +4,33 @@ import net.thechance.dukan.entity.Dukan
 import org.springframework.data.domain.Page
 import org.springframework.data.domain.Pageable
 import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.data.jpa.repository.Query
 import org.springframework.stereotype.Repository
 import java.util.UUID
 
 @Repository
 interface DukanRepository : JpaRepository<Dukan, UUID> {
     fun existsByName(name: String): Boolean
     fun existsByOwnerId(ownerId: UUID): Boolean
-    fun findByOwnerId(ownerId: UUID): Dukan?
-
+    fun findByOwnerId(ownerId: UUID): Dukan
     fun findAllByCategoriesId(categoryId: UUID, pageable: Pageable): Page<Dukan>
+    @Query(
+        """
+    SELECT DISTINCT d
+    FROM Dukan d
+    WHERE d.status = net.thechance.dukan.entity.Dukan.Status.APPROVED
+    AND EXISTS (
+        SELECT s FROM DukanShelf s
+        WHERE s.dukan = d
+    )
+    AND EXISTS (
+        SELECT p FROM DukanProduct p
+        WHERE p.dukan = d
+    )
+    ORDER BY d.createdAt DESC
+    """
+    )
```

Comment:
We dont need to check whether the Dukan has shelves since having products means it already has shelves
```suggestion
SELECT DISTINCT d
FROM Dukan d
WHERE d.status = net.thechance.dukan.entity.Dukan.Status.APPROVED
  AND EXISTS (
    SELECT 1
    FROM DukanProduct p
    WHERE p.dukan = d
  )
ORDER BY d.createdAt DESC

```

Reply:
I thought about this two and discussed it with haider and yousof but how about if the proudct hase been added wrongly with out shelfs

Reply:
then just update you checks to 
```suggestion
    @Query(
        """
    SELECT DISTINCT d
    FROM Dukan d
    WHERE d.status = net.thechance.dukan.entity.Dukan.Status.APPROVED
    AND EXISTS (
        SELECT 1 
        FROM DukanShelf s
        WHERE s.dukan = d
    )
    AND EXISTS (
        SELECT 1
        FROM DukanProduct p
        WHERE p.dukan = d
    )
    ORDER BY d.createdAt DESC
    """
    )
```